### PR TITLE
Add tests for by_two function and improve

### PR DESCRIPTION
**Pull request by M3rs**
Add basic tests for by_two function: empty input, even and odd length input.

Refactor by_two function to return more explicit namedtuple with items (url, data).

### CODE REVIEW

File: tests.py [hunk 1]
```diff
@@ -3,6 +3,41 @@
 import vanity
 
 
+class test_by_two(unittest.TestCase):
```
Comment (hugovk) [inline tests.py]: Use [CapWords convention](https://www.python.org/dev/peps/pep-0008/#class-names) for class names.
-> `TestByTwo`

I also mistakenly asked a previous contributor to put `test_normalize`, but please could you also make that `TestNormalize`? Thanks!

File: tests.py [hunk 2]
```diff
@@ -3,6 +3,41 @@
...
+    def test_odd_input(self):
+        input = iter(['test.com', 'test data',
+                      'foo.org', 'foo data',
+                      'bar.net'])
+
+        result = {url: data for (url, data) in vanity.by_two(input)}
+
+        self.assertEqual(result['test.com'], 'test data')
+        self.assertEqual(result['foo.org'], 'foo data')
+        self.assertEqual(result.get('bar.net'), None)
```
Comment (hugovk) [inline tests.py]: Let's use `assertIsNone` for this last one. 
https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertIsNone

File: tests.py (hunk 1 above)
Comment (M3rs) [inline tests.py]: Will do!  (was just trying to follow convention).

Comment (hugovk) [inline tests.py]: Thanks, it's generally good to follow the file's local convention :)

But better I didn't get it wrong in the first place!

**Comment by M3rs** [comment]
Pros/Cons of namedtuple:

(could probably rename it to something like release_info, based on use)

Code in count_downloads could look more like:

```python
for release_info in get_release_info([package], json=json):
    for url in release_info.urls:
        # etc
```

Here is a talk by Raymond Hettinger (https://www.youtube.com/watch?v=wf-BqAjZb8M), towards the end he talks about how nametuples are great. 

Pros:
- Adds clarity (each time it is used, the values urls, data are pulled from it)
- tuples are immutable (before was returning a list)
- provides access fields vs unpacking / indexing

Cons:
- Admittedly, I am having trouble finding any.

**Comment by hugovk** [comment]
Ha, I was just watching that video and had it paused! namedtuple bit is at [here](https://www.youtube.com/watch?v=wf-BqAjZb8M&t=44m48s) and makes sense.

Would you like to resolve the merge conflict?

**Comment by M3rs** [comment]
Resolved merge conflicts

**Comment by hugovk** [comment]
Thanks!
